hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -339,6 +339,11 @@ public void initializeCertificateClient(OzoneConfiguration config)
     LOG.info("Initializing secure Datanode.");
 
     CertificateClient.InitResponse response = dnCertClient.init();
+    if (response.equals(CertificateClient.InitResponse.REINIT)) {
```

Comment:
Could this be put into the below switch, as one of case?

Reply:
Unfortunately not, the rest of the switch cases rely on this reinitialization.

Reply:
Not really...
As in this case we need to redo the initialization step in a new instance that loads the certificate data from disk again and instantiates with a clean internal state.
If I create the new instance within the switch and call init, I need to get to the same switch case again, with the result of the new initialization, and I can not imagine a way of doing so within the switch case.